src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1181 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Symbols;
+
+[CompilerTrait(CompilerFeature.RequiredMembers)]
+public class RequiredMembersTests : CSharpTestBase
+{
+    private const string RequiredMemberAttribute = @"
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Field | AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+    public class RequiredMemberAttribute : Attribute
+    {
+        public RequiredMemberAttribute()
+        {
+        }
+    }
+}
+";
+
+    private CSharpCompilation CreateCompilationWithRequiredMembers(CSharpTestSource source, CSharpParseOptions? parseOptions = null, CSharpCompilationOptions? options = null, TargetFramework tfm = TargetFramework.NetCoreApp)
+        => CreateCompilation(new[] { source, RequiredMemberAttribute }, options: options, parseOptions: parseOptions);
+
+    private Action<ModuleSymbol> ValidateRequiredMembersInModule(string[] memberPaths)
+    {
+        return module =>
+        {
+            foreach (var memberPath in memberPaths)
+            {
+                var member = module.GlobalNamespace.GetMember(memberPath);
+                Assert.True(member is PropertySymbol or FieldSymbol, $"Unexpected member symbol type {member.Kind}");
+                Assert.True(member.IsRequired());
+            }
+        };
+    }
+
+    [Fact]
+    public void InvalidModifierLocations()
```

Comment:
Let's include a test for `abstract required` combination (allowed)

Reply:
Thanks. Btw, I added a note to test plan to come up with some canonical order for `required` modifier